From fc97c967bcaed18321270f118c6fca9ea0f71d96 Mon Sep 17 00:00:00 2001
From: Mingcong Bai <jeffbai@aosc.io>
Date: Wed, 30 Jul 2025 10:48:24 +0800
Subject: [PATCH 3/3] AOSCOS: fix(go): Append -L/usr/lib to fix CGO linkage

Signed-off-by: Mingcong Bai <jeffbai@aosc.io>
---
 gcc/go/lang-specs.h | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/gcc/go/lang-specs.h b/gcc/go/lang-specs.h
index 7df1b62a33e..0f3e5565c21 100644
--- a/gcc/go/lang-specs.h
+++ b/gcc/go/lang-specs.h
@@ -21,5 +21,5 @@ along with GCC; see the file COPYING3.  If not see
    for the Go language.  */
 
 {".go",  "@go", 0, 1, 0},
-{"@go",  "go1 %i %(cc1_options) %{I*} %{L*} %D %{!fsyntax-only:%(invoke_as)}",
+{"@go",  "go1 %i %(cc1_options) %{I*} %{L*} -L/usr/lib %{!fsyntax-only:%(invoke_as)}",
     0, 1, 0},
-- 
2.50.1

